README.md

```diff
@@ -1 +1,430 @@
+
+```shell
+docker compose -f dev/docker-compose-silab.yml up -d
+```
+
+Alternatively, you can set up a fully local environment with a NetBox and ISDuBa installation:
+
+```shell
+docker compose -f dev/docker-compose.yml up -d
+```
+
+If you use the fully local environment, the API token for NetBox is printed by the `netbox-setup` container. You can read it with:
+
+```shell
+docker compose -f dev/docker-compose.yml logs netbox-setup
+```
+The API token is printed to the console.
+
+To configure the netbox fetcher plugin, copy the file [assets/plugin_configs/data_source/asset/sample/netbox.toml](assets/plugin_configs/data_source/asset/sample/netbox.toml) to `assets/plugin_configs/data_source/asset/netbox_local.toml`.
+The file can be named any way you like, but it must be a toml file.
+Replace the `api_token` with the API token printed by the netbox-setup container.
+Replace the `url` with the url of the netbox instance (http://netbox.localhost/).
+
+To configure the ISDuBa fetcher plugin, copy the file [assets/plugin_configs/data_source/csaf/sample/isduba.toml](assets/plugin_configs/data_source/csaf/sample/isduba.toml) to `assets/plugin_configs/data_source/csaf/isduba_local.toml`.
+The file can be named any way you like, but it must be a toml file.
+Replace the `url` with the url of the ISDuBa instance http://isduba.localhost/.
+Replace the `username` and `password` with the credentials of the ISDuBa user (user/user).
+Replace the `keycloak_url` with the url of the keycloak instance http://keycloak.localhost/.
+
+The ISDuBa interface can be reached at http://isduba.localhost/ and the netbox interface at http://netbox.localhost/.
+The default credentials are:
+- Netbox: admin / admin
+- ISDuBa: user / user
+
+Before starting the synchronizers, make sure to create some assets and csaf advisories in the netbox and ISDuBa instances.
+Follow the corresponding instructions in the netbox and ISDuBa documentation.
+
+Next, install the python dependencies with uv by running in your terminal or inside pycharm (double tap `<Ctrl>` and enter the
+command) to set up the local python environment:
+
+```shell
+uv sync --all-extras
+```
+
+This will install all the available plugins in the `plugins/` directory in addition to the base package.
+If you want to install only the base package, just run `uv sync`.
+The plugins can be installed with `uv sync --extra <PLUGIN_NAME>` later on.
+To install multiple extras, provie multiple `--extra` arguments.
+
+### Setting up git pre-commit hooks
+
+Before commiting anything, make sure you have set up your git pre-commit hooks correctly.
+To do so, simply run the following:
+
+```shell
+uv run pre-commit install
+```
+
+To manually run the pre-commit hooks (in case you want to check without commiting), run the following:
+
+```shell
+uv run pre-commit run --all-files
+```
+
+If you want to run the linter manually, run the following:
+
+```shell
+uv run ruff check
+```
```

Comment:
This should go into the Development instructions. Is commiting via the Web-GUI in GitHub still allowed?